src/components/ScheduleConfirmationModal/index.tsx

```diff
@@ -0,0 +1,118 @@
+import Modal from '../modal';
+import {
+  HeaderTypography,
+  SubTypography,
+  DoneTypography,
+  NotDoneTypography,
+  Card,
+  ScheduleContainer,
+  ButtonContainer,
+  ElementContainer,
+  ConfirmationButtonContainer,
+  ConfirmationButton,
+  NegationButton,
+  FinalButton,
+  SchedulesOpen,
+} from './styles';
+import { Typography } from '@mui/material';
+import { TScheduleEnding } from '../../service/requests/useGetSchedulesEndingRequest/types';
+import { CheckCircleRounded, CancelRounded } from '@mui/icons-material';
+
+type Props = {
+  isOpen: boolean;
+  scheduleState?: TScheduleEnding[];
+  isSuccess: boolean;
+  isLoadingUpdate: boolean;
+  errorUpdate?: string;
+  numberScheduleOpens: number;
+  handleClickDone(schedule: TScheduleEnding): void;
+  handleClickNotDone(schedule: TScheduleEnding): void;
+  handleClose(): void;
+};
+
+const ScheduleConfirmationModal = ({
+  isOpen,
+  scheduleState,
+  handleClose,
+  handleClickDone,
+  handleClickNotDone,
+  isSuccess,
+  isLoadingUpdate,
+  errorUpdate,
+  numberScheduleOpens,
+}: Props) => {
+  const renderContent = () => {
+    return (
+      <Card>
+        <HeaderTypography>Confirme sua monitoria</HeaderTypography>
+        <SubTypography>
+          Abaixo estarão listados os eventos que já ocorreram de acordo com sua
+          agenda, confirme se a monitoria foi realizada ou não clicando no botão
+          ao lado.
+        </SubTypography>
+        <SchedulesOpen>
+          {scheduleState?.map((schedule) => (
+            <ScheduleContainer>
+              <ElementContainer>
+                <Typography variant="caption">Aluno</Typography>
+                <Typography variant="subtitle1">{schedule.name}</Typography>
+              </ElementContainer>
+              <ElementContainer>
+                <Typography variant="caption">Data</Typography>
+                <Typography variant="subtitle1">{schedule.date}</Typography>
+              </ElementContainer>
+              <ElementContainer>
+                <Typography variant="caption">Horário</Typography>
+                <Typography variant="subtitle1">{`${schedule.startHour} até ${schedule.endHour}`}</Typography>
+              </ElementContainer>
+              <ConfirmationButtonContainer>
+                {schedule.status === 2 ? (
+                  <>
+                    <NegationButton
+                      loading={isLoadingUpdate}
+                      onClick={() => handleClickNotDone(schedule)}
+                    >
+                      Não realizada
+                    </NegationButton>
+                    <ConfirmationButton
+                      loading={isLoadingUpdate}
+                      onClick={() => handleClickDone(schedule)}
+                    >
+                      Realizada
+                    </ConfirmationButton>
+                  </>
```

Comment:
O problema do loading é que quando uma delas tá carregando, todos os botões recebem a prop de loading como true, sendo que só o card clicado deveria receber true. Uma sugestão é:
1. Salva em um estado (`selectedSchedule`) qual foi o agendamento selecionado nas funções `handleClickDone` e `handleClickNotDone` do hook.
2. Daí aqui você só vai setar o loading se for o agendamento que está sendo carregado.
3. Importante também setar o disabled para todos os botões quando algum estiver carregando.

No fim vai ficar assim:
```suggestion
                  <>
                    {selectedSchedule.id === schedule.id && isLoadingUpdate ? (
                        <LoadingIcon />
                      ) : (
                        <NegationButton
                          disabled={isLoadingUpdate}
                          onClick={() => handleClickNotDone(schedule)}
                        >
                          Não realizada
                        </NegationButton>
                        <ConfirmationButton
                          disabled={isLoadingUpdate}
                          onClick={() => handleClickDone(schedule)}
                        >
                          Realizada
                        </ConfirmationButton>
                      )
                    }
                  </>
```

Isso vai ficar um pouco diferente do Figma mas conversei com o Chris para ser assim pq é mais fácil de fazer do que a proposta original.